File: Binary Tree Level Order Traversal.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int> > levelOrder(TreeNode *root) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        vector<vector<int>> result;
        
        if(root!=NULL)
        {
            int cur = 1;
            int next = 0;
            vector<int> oneResult;
            list<TreeNode *> nodeList;
            nodeList.push_back(root);
            bool reverse = true;
            
            while(!nodeList.empty())
            {
                TreeNode *curNode = NULL;
                cur--;
         
                curNode = nodeList.front();
                nodeList.pop_front();
        
                if(curNode->left)
                {
                    nodeList.push_back(curNode->left);
                    next++;
                }
                
                if(curNode->right)
                {
                    nodeList.push_back(curNode->right);
                    next++;
                }
                
         
                oneResult.push_back(curNode->val);
                if(cur==0)
                {
                    result.push_back(oneResult);
                    oneResult.clear();
                    cur = next;
                    next = 0;
                }
            }
            
        }
        
        return result; 
    }
};

File: Binary Tree Level Order Traversal.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ArrayList<ArrayList<Integer>> levelOrder(TreeNode root) {
        // Start typing your Java solution below
        // DO NOT write main() function
        ArrayList<ArrayList<Integer>> ret = new ArrayList<ArrayList<Integer>>();
        getLevelOrder(root, 1, ret);
        return ret;
    }
    
    private void getLevelOrder(TreeNode root, int level, ArrayList<ArrayList<Integer>> ret)
    {
        if(root == null)
        {
            return;
        }
        
        if(level > ret.size())
        {
            ret.add(new ArrayList<Integer>());
        }
        ret.get(level-1).add(root.val);
        getLevelOrder(root.left, level + 1, ret);
        getLevelOrder(root.right, level + 1, ret);
    }
    
}